packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -7,39 +7,78 @@ import { ITelemetryLogger, ITelemetryBaseEvent } from "@fluidframework/common-de
 import { TelemetryLogger } from "./logger";
 
 /**
- * The MockLogger records events sent to it, and then can walk back over those events
+ * Logger that records events sent to it, and then can walk back over those events
  * searching for a set of expected events to match against the logged events.
+ *
+ * @internal
  */
 export class MockLogger extends TelemetryLogger implements ITelemetryLogger {
```

Comment:
i think we should deprecate it. we shouldn't ship test utils inline with non-test source. then is next move it to test.

Reply:
My initial intention was to move it to test-utils but that ends up causing circular dependencies in some packages that currently use this class, when those packages try to import it from test-utils. Same thing when I tried to move it to test-runtime-utils. (Trying to resolve that is how I ran into the unused dependencies in the other PR you saw). Addressing the circular dependencies might be possible but didn't seem trivial, so I thought an acceptable first step was to make it internal and remove it from our public API surface.

Reply:
totally fair. can you please file an issue to get this removed from our public exports entirely? In general, i don't really trust internal, they still get exported, i don't think tooling like vs code lets you know if you are using something internal, and modifications can still cause problems for consumers if they mix packages.

Reply:
Sure. However, by "external exports" you mean out of a @fluidframework/ package, right? We'd still put it in a @fluid-tools or similar to be able to share it across the repo but make it clearer that it isn't meant for external consumers?

Reply:
[ADO item to move MockLogger to a better location](https://dev.azure.com/fluidframework/internal/_workitems/edit/3726)

Reply:
see: https://github.com/microsoft/FluidFramework/blob/main/packages/dds/merge-tree/src/test/index.ts

sequence consumes it (in repo), but it not published as we trim the tests from the output before publish, so its never externally exported.

Reply:
there may also be an @fluid-internal package it could fit in so its not externally published.

Reply:
In this case it might require a package, because e2e tests use the MockLogger, so when they run in their own pipeline with access to only the published artifacts, they wouldn't be able to see it. (Technically, we do publish src/test and dist/test as artifacts as well, but what we do today with that already feels dirty; having e2e tests use it from there feels even ickier IMO 😄).